compiler/pom.xml

```diff
@@ -32,6 +32,16 @@
       <artifactId>auto-value</artifactId>
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>org.jgrapht</groupId>
```

Comment:
[maven-shade-plugin](https://maven.apache.org/plugins/maven-shade-plugin/) creates a “fat jar” (embedding dependencies) such that the annotation processor JAR is self-contained (except for Dagger, which should generally be on the processor path already anyway, and Guava, because of Guava's API stability guarantees, and because Dagger doesn't “shade” Guava either).
For instance, that makes sure we won't run into conflicts if we use `auto-common` 0.4 but another annotation processor in the processor path uses `auto-common` 0.5 or 0.6 (which, as we've seen in another PR, has breaking API changes): all `auto-common` classes are renamed (relocated) into a `bullet.thirdparty` subpackage so they become “unique” to this JAR and cannot conflict with others (at the price of having the classes several times with different names, i.e. bigger JARs)